roles/k8s-cluster/all-nodes/tasks/main.yaml

```diff
@@ -16,32 +18,40 @@
     - software-properties-common
 
 - name: Add an apt signing key for Docker
+  tags:
+  - docker
   become: yes
   apt_key:
     url: https://download.docker.com/linux/ubuntu/gpg
     state: present
 
 - name: Add apt repository for stable version
+  tags:
+  - docker
   become: yes
   apt_repository:
-    repo: deb [arch=amd64] https://download.docker.com/linux/ubuntu xenial stable
+    repo: deb [arch=amd64] https://download.docker.com/linux/ubuntu {{ ansible_facts.lsb.codename }} stable
     state: present
 
 - name: Install docker and its dependecies
+  tags:
+  - docker
   become: yes
   apt:
     name: "{{ packages }}"
     state: present
     update_cache: yes
   vars:
     packages:
-    - docker-ce
-    - docker-ce-cli
-    - containerd.io
+    - docker-ce=5:19.03.4~3-0~ubuntu-{{ ansible_facts.lsb.codename }}
```

Comment:
Maybe it could stay for 19.03, so minor bugs would be fixed.

Reply:
I agree with you, however the k8s17 documentation specifies the docker version 19.03.4
`https://v1-17.docs.kubernetes.io/docs/setup/production-environment/container-runtimes/#docker`

would it be better to change to version 19.03?

Reply:
I don't think so, if he recommends it.
@andersonluisribeiro

Reply:
Kubernetes release notes says it was validated in version 19.03:

https://v1-17.docs.kubernetes.io/docs/setup/release/notes/

This is the PR for kubernetes repository:

https://github.com/kubernetes/kubernetes/pull/84476

The docker 19.03.4 was released in 2019-10-17. The PR was created on 2019-10-28. I think this is the reason why they put this version in the documentation. The version 19.03.5 of the Docker was released in 2019-11-14.

Reply:
However I think we can use version 19.03 of the Docker.

Reply:
I don't think we have this option of installing the docker in version 19.03

Following the docker documentation, either you install on the latest version or a specific version.

So, the solution adopted was to fix in version 19.03.14 which is the last bugfix of 19.03